meter/dsmr.go

```diff
@@ -137,10 +139,13 @@ func NewDsmr(uri, energy string, timeout time.Duration) (api.Meter, error) {
 // based on https://github.com/basvdlei/gotsmart/blob/master/gotsmart.go
 func (m *Dsmr) run(conn *bufio.Reader, done chan struct{}) {
 	log := util.NewLogger("dsmr")
+	backoff := backoff.NewExponentialBackOff()
+	backoff.InitialInterval = time.Second
+	backoff.MaxInterval = 5 * time.Minute
 
 	handle := func(op string, err error) {
 		log.ERROR.Printf("%s: %v", op, err)
-		if errors.Is(err, net.ErrClosed) {
+		if err == io.EOF || errors.Is(err, net.ErrClosed) {
```

Comment:
Maybe add `io.ErrUnexpectedEOF`, too for good measure? Really not sure what the best practice is here. Or is the `net.ErrClosed` already sufficient?

Reply:
addressed using this guidance: https://github.com/golang/go/wiki/ErrorValueFAQ#how-should-i-change-my-error-handling-code-to-work-with-the-new-features